[#2460] Reuse methods for processing Kafka headers

kaniyan opened the pull request:
This is for issue #2460

kaniyan commented:
@b-abel As we discussed, I have raised this PR for issue #2460. Would you mind also taking a look?

calohmn commented:
You could replace `new KafkaHeaderImpl` by `KafkaHeader.header` in `KafkaMessageHelper`.
Do you want to do the suggested renaming to `RecordHelper` also here as a 2nd commit?

calohmn commented on clients/kafka-common/src/main/java/org/eclipse/hono/client/kafka/KafkaMessageHelper.java:
```diff
@@ -199,4 +202,18 @@ public static boolean isTtlElapsed(final List<KafkaHeader> headers) {
+    /**
+     * Gets the String value of a specific <em>header</em>.
+     * <p>
+     * If the headers contains multiple occurrences of the same key, the value of its first
```
"contain" (also in `getHeaderValue`)

kaniyan commented:
> You could replace `new KafkaHeaderImpl` by `KafkaHeader.header` in `KafkaMessageHelper`.

It was already there in the existing code and it is good to replace it with `KafkaHeader.header`

kaniyan commented:
> Do you want to do the suggested renaming to `RecordHelper` also here as a 2nd commit?

FMPOV `RecordHelper` seems to be quite generic. How about `KafkaRecordHelper`?

calohmn commented:
> How about KafkaRecordHelper?

I'm fine with that.

kaniyan commented:
> Do you want to do the suggested renaming to `RecordHelper` also here as a 2nd commit?

@calohmn I have added a 2nd commit regarding renaming of that file and also updated in accordance with the above comments.

calohmn reviewed (approved):
LGTM

b-abel commented on clients/kafka-common/src/main/java/org/eclipse/hono/client/kafka/KafkaRecordHelper.java:
```diff
@@ -199,4 +201,18 @@ public static boolean isTtlElapsed(final List<KafkaHeader> headers) {
+     * @return The value or {@code null} if the headers do not contain a value of the expected type for the given key.
+     * @throws NullPointerException if any of the parameters is {@code null}.
+     */
+    public static String getStringHeaderValue(final List<KafkaHeader> headers, final String key) {
```
Why does this not return an optional like the other methods here?

kaniyan replied:
This method has been moved from the `RecordUtil` class (as that class has been deleted). The classes which use this utility method expect that this method returns a `String` value if present else _null_.

b-abel replied:
I know that. But should not `KafkaRecordHelper` define its interface and code that uses it be adapted instead?

kaniyan replied:
FMPOV it is more of a taste if to return a _String_ or an _Optional_ for a utility method. Considering that we already return _Optional_ for an already existing method in that class, I find it reasonable to have the same style. If `getStringHeaderValue(...)` is to return an _Optional_, I find it there is not much value in having that method and rather better to use the `getHeaderValue(headers, key, String.class)`.  I have made changes to use the  `getHeaderValue(headers, key, String.class)`.

b-abel commented:
LGTM

kaniyan commented:
The conflicts due to merge of PR #2498 has been resolved.
